packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -42,20 +41,17 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
     @inject(EditorManager) protected readonly editorManager: EditorManager;
 
     onStart(app: FrontendApplication): void {
-        this.outlineViewService.onDidChangeOpenState(async open => {
-            if (open) {
-                this.toDisposeOnClose.push(this.toDisposeOnEditor);
-                this.toDisposeOnClose.push(DocumentSymbolProviderRegistry.onDidChange(
-                    debounce(() => this.updateOutline())
-                ));
-                this.toDisposeOnClose.push(this.editorManager.onCurrentEditorChanged(
-                    debounce(() => this.handleCurrentEditorChanged(), 50)
-                ));
-                this.handleCurrentEditorChanged();
-            } else {
-                this.toDisposeOnClose.dispose();
-            }
-        });
+
+        // updateOutline and handleCurrentEditorChanged need to be called even when the outline view widget is closed
+        // in order to udpate breadcrumbs.
```

Comment:
i don't understand it, if an editor is closed there should not be breadcrumbs.

Reply:
The removed condition does not check if an _editor_ is open or closed it checks if the _outline view_ is open or closed. Previously, the outline changes are propagated when the outline view is open only. Now we need the information even when the outline view is closed because the outline view is not the only one that uses this information anymore. The breadcrumbs need the information as well.

What we could do is to check whether the outline view is open _OR_ the breadcrumbs are visible (by the preferences entry `breadcrumbs.enabled`). I don't know if it is really worth it.